File: solutions/lua/triangle/1/triangle.lua
local triangle = {}


function triangle.kind(a, b, c)

    assert(a > 0 and b > 0 and c > 0, 'Input Error')
    assert(a + b > c, 'Input Error')
    assert(a + c > b, 'Input Error')
    assert(b + c > a, 'Input Error')

    if a == b and b == c then
        return 'equilateral'
    elseif a == b or a == c or b == c then
        return 'isosceles'
    else
        return 'scalene'
    end

end

return triangle
    

File: solutions/go/triangle/1/triangle.go
// Package triangle provides operations to test the properties of a triangle.
package triangle


// Kind represents a kind of triangle
type Kind int

const (
    NaT Kind = iota // not a triangle
    Equ // equilateral
    Iso // isosceles
    Sca // scalene
)

// KindFromSides should have a comment documenting it.
func KindFromSides(a, b, c float64) Kind {
    if a <= 0 || b <= 0 || c <= 0 || (a+b) < c || (a+c) < b || (b+c) < a {
        return NaT
    }
	if a == b && b == c {
        return Equ
    }
    if a == b || b == c || a == c {
        return Iso
    }
    return Sca
}
